### PR TITLE
[Bug] GravatarImageSource throws exception when offline

### LINKED ISSUE #1770: [Bug] GravatarImageSource throws exception when offline
### Description

If a GravatarImageSource is used when the app is offline, such as in flight mode, and it has no cached image to display, it throws the exception: `Java.Net.UnknownHostException: 'Unable to resolve host "www.gravatar.com": No address associated with hostname'`. If the image has already been cached then this problem does not occur.

### Stack Trace

No stack trace.

### Link to Reproduction Sample

https://github.com/varyamereon/GravatarSample

### Steps to Reproduce

1. Run the repo or simply add a GravatarImageSource
2. Run the app on a device with no internet connection
3. Exception is thrown

### Expected Behavior

Should fail gracefully, in my example the GravatarImageSource is used in an AvatarView, and I would expect the avatar view to simply show the Text value until the device has connectivity restored and can obtain the Gravatar.

### Actual Behavior

Exception thrown and app terminates.

### Basic Information

- Version with issue: 1.3.1
- Last known good version: n/a
- IDE: Visual Studio 2022 Windows
- Platform Target Frameworks:
  - iOS:  not tested
  - Android: 11
  - UWP:  not tested
- Android Support Library Version: n/a
- Nuget Packages: n/a
- Affected Devices: Tested on OnePlus 6

### Workaround
Haven't found a workaround. Have tried setting the source in code behind wrapped in a try/catch statement, but the exception is still thrown and the app terminates.

### PR DESCRIPTION
**Pull request by VladislavAntonyuk**
GravatarImageSource throws exception when offline
Fix Nullable

### Issues Fixed ###

- Fixes #1770

### PR Checklist ###

- [ ] Has a linked Issue, and the Issue has been `approved`
- [ ] Has tests (if omitted, state reason in description)
- [ ] Has samples (if omitted, state reason in description)
- [ ] Rebased on top of main at time of PR
- [ ] Changes adhere to coding standard

- [ ] Updated [documentation](https://github.com/MicrosoftDocs/xamarin-communitytoolkit)

### CODE REVIEW

File: src/CommunityToolkit/Xamarin.CommunityToolkit.UnitTests/Xamarin.CommunityToolkit.UnitTests.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netcoreapp2.1;netcoreapp3.1;net461</TargetFrameworks>
```
Comment (VladislavAntonyuk) [inline src/CommunityToolkit/Xamarin.CommunityToolkit.UnitTests/Xamarin.CommunityToolkit.UnitTests.csproj:4]: fix build out of the box. netcoreapp2.1 is the end of support

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/StatusBar/PlatformStatusBarEffect.ios.cs [hunk 2]
```diff
@@ -99,9 +99,9 @@ static void UpdateStatusBarAppearance()
 			}
 		}
 
-		static void UpdateStatusBarAppearance(UIWindow window)
+		static void UpdateStatusBarAppearance(UIWindow? window)
```
Comment (VladislavAntonyuk) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/StatusBar/PlatformStatusBarEffect.ios.cs:102]: see line 97. window can be null

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Views/GravatarImageSource/Shared/GravatarImageSourceHandler.android.ios.macos.tizen.uwp.wpf.cs [hunk 3]
```diff
@@ -87,12 +88,20 @@ static string CacheFileName(GravatarImageSource gis, float scale)
...
+			{
+				using var response = await lazyHttp.Value.GetAsync(requestUri);
 
-			if (!response.IsSuccessStatusCode)
-				return Array.Empty<byte>();
+				if (!response.IsSuccessStatusCode)
+					return Array.Empty<byte>();
 
-			return await response.Content.ReadAsByteArrayAsync();
+				return await response.Content.ReadAsByteArrayAsync();
```
Comment (bijington) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Views/GravatarImageSource/Shared/GravatarImageSourceHandler.android.ios.macos.tizen.uwp.wpf.cs:98]: Can this return `null`?

File: samples/XCT.Sample.sln [hunk 4]
```diff
@@ -1,6 +1,6 @@
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio Version 16
-VisualStudioVersion = 16.0.30204.135
+# Visual Studio Version 17
```
Comment (pictos) [inline samples/XCT.Sample.sln:2]: Do we need to update this file?

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Views/GravatarImageSource/Shared/GravatarImageSourceHandler.android.ios.macos.tizen.uwp.wpf.cs (hunk 3 above)
Comment (pictos) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Views/GravatarImageSource/Shared/GravatarImageSourceHandler.android.ios.macos.tizen.uwp.wpf.cs:98]:
```suggestion
				return await response.Content.ReadAsByteArrayAsync().ConfigureAwait(false);
```

**Review by pictos — COMMENTED** [review]
Looks good in general, just some observations

File: samples/XCT.Sample.sln (hunk 4 above)
Comment (VladislavAntonyuk) [reply samples/XCT.Sample.sln:2]: VS updated it automatically. Let's keep VS version 16.

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Views/GravatarImageSource/Shared/GravatarImageSourceHandler.android.ios.macos.tizen.uwp.wpf.cs (hunk 3 above)
Comment (VladislavAntonyuk) [reply src/CommunityToolkit/Xamarin.CommunityToolkit/Views/GravatarImageSource/Shared/GravatarImageSourceHandler.android.ios.macos.tizen.uwp.wpf.cs:98]: looks like no.
image

**Review by pictos — APPROVED** [review]

**Pull request merged**